lib/guard/puma/runner.rb

```diff
@@ -43,11 +43,21 @@ def start
     end
 
     def halt
-      run_puma_command!("halt")
+      Net::HTTP.get build_uri('halt')
+
+      return true
+    rescue Errno::ECONNREFUSED
+      # server may not have been started correctly, but we are halting so who cares.
+      return true
```

Comment:
(1) it's less code and (2) the logic of what's going on is more explicit and more easily followed. I prefer this implementation, but if you like I will update it.